core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -775,6 +775,24 @@ class DynamicBrokerReconfigurationTest extends ZooKeeperTestHarness with SaslSet
     verifyMarkPartitionsForTruncation()
   }
 
+  @Test
+  def testDefaultValueRestoredAfterDeleteDynamicConfig(): Unit = {
+    val newProps = new Properties
+    newProps.put(KafkaConfig.LogRetentionTimeMillisProp, "100000")
+    newProps.put(KafkaConfig.LogFlushIntervalMsProp, "10000")
+    TestUtils.incrementalAlterConfigs(servers, adminClients.head, newProps, perBrokerConfig = false).all.get
```

Comment:
Shouldn't we also `waitForConfigOnServer` after the first alteration of the config?

Reply:
Thanks for the comments. We could add but not necessary since we care about the fact that the value for broker-default level should be overwritten by that for broker-level. Does it make sense?

Reply:
That makes sense, thanks.